packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -373,6 +381,20 @@ class CameraController extends ValueNotifier<CameraValue> {
     }
   }
 
+  /// Sets the description while the camera is recording.
+  Future<void> setDescriptionWhileRecording(
+      CameraDescription description) async {
```

Comment:
The high-level API of `CameraController` seems somewhat confusing now. Before, `CameraDescription` was final so you had to pick a camera before creating the controller. Now, you can either set it before, or you can change it while recording video, but *at no other time*. That seems very counterintuitive; why is it only possible to change cameras at the hardest possible time? If we're allowing changing cameras after the creation of `CameraController`, which not allow changing it at any time, and have the special handling of changing it during recording be an internal implementation detail?

Reply:
Thats a very good point. I will make it be setDescription then handle the differences internally